classes/passwd.oeclass

```diff
@@ -19,18 +19,14 @@ GROUP_FILES ?= "${SRCDIR}/group"
 PASSWD_POSTFUNC:machine = " do_install_passwd"
 do_install[postfuncs] += "${PASSWD_POSTFUNC}"
 do_install_passwd () {
-	i=0
 	for f in ${PASSWD_FILES} ; do
-		mkdir -p ${D}/${passwddir}
-		let i=$i+1
-		cp $f ${D}/${passwddir}/${PN}.$i
+		mkdir -p ${D}/${passwddir}/${PN}
+		cp $f ${D}/${passwddir}/${PN}/
```

Comment:
How about this: We place the passwd files from a given recipe under ${passwd}/${PN}/ as above. We use cp -n to cause an error if two files would collide. If for some reason one has two files with the same basename and that can't easily be changed, a PASSWD_FILES entry can have an optional ":where/to/put/it" appended. So your example would be `PASSWD_FILES = "${SRCDIR}/foo/my-file:foo/my-file ${SRCDIR}/bar/my-file:bar/my-file"`. We already have a few places where we pass multiple pieces of information as a colon seperated list (e.g. some AUTO_PACKAGE_LIBS vars), and rely on nobody sane putting a : in a path. No current recipes need this AFAICT, and when the passwd files come from a file:// it's probably always easier to just follow the avahi example and use ${packagename}-passwd,